### ISSUE TITLE
BSS initialization in a wrong place for "plat-sunxi"

### ISSUE THREAD

**Issue by P1119r1m** [OP-TEE/optee_os#1607]
Branch:
optee/master

Commit:
432f64c

File:
core/arch/arm/plat-sunxi/entry.S

Problem:
BSS initialization executed after the initialization of the MMU table (global variable array "static_memory_map[]"):
```
LOCAL_FUNC reset , :
	...
	bl	core_init_mmu_map        <-------- Here is static_memory_map[] initialized!
	bl	core_init_mmu_regs
	bl	cpu_mmu_enable
	bl	cpu_mmu_enable_icache
	bl	cpu_mmu_enable_dcache

	/* init BSS section */
	ldr     r0, =__bss_start
	ldr     r2, =__bss_end
	sub     r2, r2, r0
	ldr     r1, =0
	bl      memset
	...
```

Solution:
Provide BSS initialization before static_memory_map[] initialized by core_init_mmu_map:
```
LOCAL_FUNC reset , :
	...
	/* init BSS section */
	ldr     r0, =__bss_start
	ldr     r2, =__bss_end
	sub     r2, r2, r0
	ldr     r1, =0
	bl      memset

	bl	core_init_mmu_map        <-------- Here is static_memory_map[] initialized!
	bl	core_init_mmu_regs
	bl	cpu_mmu_enable
	bl	cpu_mmu_enable_icache
	bl	cpu_mmu_enable_dcache
	...
```

**Comment by jenswi-linaro** [comment]
Or even better, update the port to use `CFG_GENERIC_BOOT = y`.

**Comment by jforissier** [comment]
@P1119r1m good catch, it's my mistake. I have no way of testing `plat-sunxi` unfortunately, so mishaps  are likely to happen again in the future (especially while `CFG_GENERIC_BOOT` is not supported as @jenswi-linaro said).

Would you mind submitting a fix (pull request)? Either the one you are proposing, which looks OK, or (if you are more ambitious...) give the "generic boot" option a try?

BTW, how did you find the issue? Is it by trying to run OP-TEE on a `plat-sunxi` board, or just by browsing the code?

**Comment by P1119r1m** [comment]
@jforissier and @jenswi-linaro Thanks for your responses, guys! Mistakes will be everywhere and always. We're just people :)
I should say that I have no plat-sunxi board...
I've tried to run optee-os on the ARMv7 and took plat-sunxi platform as a base.
Further, by browsing sources I found this tricky place.
So, probably It would be better now (at least for me) simply to fix it using proposed solution, because I cannot check "CFG_GENERIC_BOOT solution" on a real hardware and there will be a lot of changes.
